packages/mds-transaction-service/service/validators.ts

```diff
@@ -14,123 +14,156 @@
  * limitations under the License.
  */
 
-import { SchemaObject, SchemaValidator } from '@mds-core/mds-schema-validators'
-import { UUID } from '@mds-core/mds-types'
+import { EventSchema } from '@mds-core/mds-ingest-service'
+import { JSONSchemaType, SchemaObject, SchemaValidator } from '@mds-core/mds-schema-validators'
+import { UUID, VEHICLE_TYPES } from '@mds-core/mds-types'
 import {
   ComplianceViolationDetailsDomainModel,
+  CurbUseDetailsDomainModel,
+  CustomReceiptDetailsDomainModel,
   FEE_TYPE,
   ReceiptDomainModel,
+  SORTABLE_COLUMN,
+  SORT_DIRECTION,
   TransactionDomainModel,
   TransactionOperationDomainModel,
   TransactionSearchParams,
   TransactionStatusDomainModel,
   TRANSACTION_OPERATION_TYPE,
-  TRANSACTION_STATUS_TYPE
+  TRANSACTION_STATUS_TYPE,
+  TripReceiptDetailsDomainModel
 } from '../@types'
 
-const uuidSchema = { type: 'string', format: 'uuid' }
+const uuidSchema = <const>{ type: 'string', format: 'uuid' }
 
-const timestampSchema = { type: 'integer', minimum: 100_000_000_000, maximum: 99_999_999_999_999 }
+const timestampSchema = <const>{
+  type: 'integer',
+  minimum: 100_000_000_000,
+  maximum: 99_999_999_999_999
+}
 
-/**
- * Currently unused, but could be!
- */
-// const { $schema: tripReceiptSchema } = SchemaValidator<TripReceiptDetailsDomainModel>({
-//   type: 'object',
-//   properties: {
-//     trip_id: uuidSchema,
-//     start_timestamp: timestampSchema,
-//     end_timestamp: timestampSchema,
-//     vehicle_type: { type: 'string', enum: [...VEHICLE_TYPES] },
-//     start_geography_id: uuidSchema,
-//     end_geography_id: { ...uuidSchema, nullable: true },
-//     duration: { type: 'integer' }, // duration of the trip in seconds
-//     distance: { type: 'integer' } // distance traveled in trip, in meters
-//   },
-//   required: [
-//     'distance',
-//     'duration',
-//     'end_geography_id',
-//     'end_timestamp',
-//     'start_geography_id',
-//     'start_timestamp',
-//     'trip_events',
-//     'trip_id',
-//     'vehicle_type'
-//   ]
-// })
+const nullableTimestamp = <const>{ ...timestampSchema, nullable: true, default: null }
+const nullableString = <const>{ type: 'string', nullable: true, default: null }
+const nullableNumber = <const>{ type: 'number', nullable: true, default: null }
+const { $id, ...TelemetrySchemaNoId } = EventSchema.properties.telemetry
+const { $id: tId, ...EventSchemaNoId } = EventSchema
+const usableEventSchema = {
+  ...EventSchemaNoId,
+  properties: { ...EventSchemaNoId.properties, telemetry: TelemetrySchemaNoId }
+}
+
+const tripReceiptSchema: JSONSchemaType<TripReceiptDetailsDomainModel> = {
+  type: 'object',
+  properties: {
+    trip_id: uuidSchema,
+    start_timestamp: timestampSchema,
+    end_timestamp: timestampSchema,
+    vehicle_type: { type: 'string', enum: [...VEHICLE_TYPES] },
+    start_geography_id: uuidSchema,
+    end_geography_id: { ...uuidSchema, nullable: true },
```

Comment:
Why is end nullable but start not nullable? Couldn't a trip end in our jurisdiction that started outside of it?

Reply:
Yeah, this is something we should probably fix (though also probably out of scope for this ticket). A trip could certainly have one or _both_ endpoints out-of-bounds.